plugins/modules/fabric_transits_workflow_manager.py

```diff
@@ -0,0 +1,1354 @@
+
+        Args:
+            self (object): An instance of a class that provides access to Cisco Catalyst Center.
+            task_id (str): The unique identifier of the task for which you want to retrieve details.
+        Returns:
+            dict or None: A dictionary status information about the specified task,
+            or None if the task with the given task_id is not found.
+        Description:
+            Call the API 'get_tasks_by_id' to get the status of the task.
+            Return the details along with the status of the task.
+        """
+
+        result = None
+        response = self.dnac._exec(
+            family="task",
+            function="get_tasks_by_id",
+            params={"id": task_id}
+        )
+        self.log('Task Details: {response}'.format(response=response), 'DEBUG')
+        self.log("Retrieving task details by the API 'get_tasks_by_id' using task ID: {task_id}, Response: {response}"
+                 .format(task_id=task_id, response=response), "DEBUG")
+
+        if response and isinstance(response, dict):
+            result = response.get('response')
+
+        return result
+
+    def check_tasks_response_status(self, response, api_name):
+        """
+        Get the site id from the site name.
+
+        Parameters:
+            self - The current object details.
+            response (dict) - API response.
+            api_name (str) - API name.
+        Returns:
+            self (object) - The current object with updated desired Fabric Transits information.
+        Description:
+            Poll the function 'get_tasks_by_id' until it returns either 'SUCCESS' or 'FAILURE'
+            state or till it reaches the maximum timeout.
+            Log the task details and return self.
+        """
+
+        if not response:
+            self.msg = "response is empty"
+            self.status = "exited"
+            return self
+
+        if not isinstance(response, dict):
+            self.msg = "response is not a dictionary"
+            self.status = "exited"
+            return self
+
+        response = response.get("response")
+        if response.get("errorcode") is not None:
+            self.msg = response.get("response").get("detail")
+            self.status = "failed"
+            return self
+
+        task_id = response.get("taskId")
+        start_time = time.time()
+        while True:
+            end_time = time.time()
+            if (end_time - start_time) >= self.max_timeout:
+                self.msg = "Max timeout of {max_timeout} sec has reached for the task id '{task_id}'. " \
+                           .format(max_timeout=self.max_timeout, task_id=task_id) + \
+                           "Exiting the loop due to unexpected API '{api_name}' status.".format(api_name=api_name)
+                self.log(self.msg, "WARNING")
+                self.status = "failed"
+                break
+
+            task_details = self.get_tasks_by_id(task_id)
```

Comment:
Can we rename task_details into task_status..as we are getting status in line 998